drivers/modem/gsm_ppp.c

```diff
@@ -829,13 +833,12 @@ static void mux_setup(struct k_work *work)
 	/* We need to call this to reactivate mux ISR. Note: This is only called
 	 * after re-initing gsm_ppp.
 	 */
-	if (IS_ENABLED(CONFIG_GSM_MUX) &&
-	    gsm->ppp_dev && gsm->state == STATE_CONTROL_CHANNEL) {
+	if (gsm->ppp_dev && gsm->state == GSM_PPP_STATE_CONTROL_CHANNEL) {
```

Comment:
I don't think removing the ifdef wrap makes much sense, unless you explicitly verified that gcc is smart enough to realize `state == CONTROL_CHANNEL` is never true.

Reply:
I don't really understand, are you referring to the removal of `IS_ENABLED(CONFIG_GSM_MUX)` in L832?
I removed it because `mux_setup` is only called if `CONFIG_GSM_MUX` is true so there isn't a need to check that again.

Reply:
Yeah. I missed that. Great!